### ISSUE TITLE
Add a link shortener for shared programs

### ISSUE THREAD

**Issue by zigdon** [robsimmons/dusa#97]
Currently, when I click the share link, I get the full buffer encoded in the link, e.g. 

`https://dusa.rocks/#jsonz=XZBBa4QwEIX_yiNeXIhhrT1Jt4feCqUU2oKCsCQmbANuFKO9bPe_l4y62uaS9-aN35i5sIHlrO2MY5zpth7PxoVKhKfe6pPxlYsg43SHBCrOwlXH6S5U5_Oz0ZqiylnfSKchxU2qVdar1CJ017KDhPVIxWRUMNls6m2iF1M5o08GksAkFYFnqdeWAiXyBKRLFKJyil5GgfW47DlSjjuOjOP-eustUBIkUqNtBuvw_PpxfHv5fEfXjL5yQzvIBkVAhAI21P8IP8h-oG30RtZfUjUhyxNMQfEnCJ9XDti08uAXIpnNsEfsaci3bKzG8lcPB4R90ZR52cX0HG3Owa34o1irBBHs-gs`

I was not expecting that, instead was hoping to get something like `https://dusa.rocks/4w52su5y`. Is that possible? Even if it was just a redirect to an external shortener (e.g. https://tinyurl.com/4w52su5y) it would be a better experience.

**Comment by robsimmons** [comment]
This is really helpful feedback, thank you!

This choice of stems from a philosophical commitment to local-first, surveillance-free software — and it also it means that links won't expire but doesn't commit us to an ever-increasing burden of supporting existing links. I wouldn't be opposed to opening a modal view that offers a shortened link (and maybe an "always shorten links" default checkbox), but I think making this a good experience would require API access to a shortnerer, at least for tinyurl as I read their terms of service — which would run us a couple hundred dollars a year. Definitely interested if you have thoughts on shorteners with free API access, or in other ways of making this experience less suprising

**Comment by zigdon** [comment]
Yeah, I like the idea of having dusa be entirely client-side. And if the shortening is via preference, it is easy to keep it client side by not enabling the service.

Could something like https://ulvis.net/developer.html work? I've no experience with them, and honestly don't even know how to go about validating such a service, but at least it's entirely free?
